src/snowcli/app/cli_app.py

```diff
@@ -1,48 +1,86 @@
 from __future__ import annotations
 
-import importlib
 import logging
 from pathlib import Path
+from typing import Optional
 
-import typer
 import click
+import typer
 
 from snowcli import __about__
+from snowcli.app.commands_registration.commands_registration_with_callbacks import (
+    CommandsRegistrationWithCallbacks,
+)
 from snowcli.app.dev.commands_structure import generate_commands_structure
+from snowcli.app.dev.docs.generator import generate_docs
+from snowcli.app.dev.pycharm_remote_debug import (
+    setup_pycharm_remote_debugger_if_provided,
+)
 from snowcli.app.main_typer import SnowCliMainTyper
 from snowcli.config import config_init, cli_config
-from snowcli.app.dev.docs.generator import generate_docs
 from snowcli.output.formats import OutputFormat
 from snowcli.output.printing import print_result
 from snowcli.output.types import CollectionResult
-from snowcli.app.dev.pycharm_remote_debug import (
-    setup_pycharm_remote_debugger_if_provided,
-)
 
 app: SnowCliMainTyper = SnowCliMainTyper()
 log = logging.getLogger(__name__)
 
+_commands_registration = CommandsRegistrationWithCallbacks()
+
+
+def _do_not_execute_on_completion(callback):
```

Comment:
Previously our completion was broken. It raised errors when trying to use flags like `--version`, because callbacks for these flags were executed and output of this callbacks was mixed with completion.

`resilient_parsing` is a flag determining completion mode (found in Typer docs).